### PR TITLE
feat: product rating review

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/browse/SFUI2-1183

# Scope of work

Product rating review block

# Screenshots of visual changes
Screenshot 2023-06-28 at 12 44 26

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  return (
+    <div className="max-w-[376px] md:max-w-[768px]">
+      <h3 className="font-bold py-2 pl-4 pr-3 typography-headline-4">Leave a review</h3>
+      <form className="grid grid-cols-[100px_1fr] py-2 px-4 gap-4 md:grid-cols-[176px_1fr] grid-rows-[100px_1fr] md:grid-rows-[28px_1fr] items-center md:items-start">
+        <img
+          src="http://localhost:3100/@assets/smartwatch.png"
+          alt="Smartwatch"
+          width="100"
+          height="100"
+          className="mx-auto border border-neutral-200 rounded-md aspect-square w-[100px] md:w-[176px]"
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx:27]:
```suggestion
          className="mx-auto border border-neutral-200 rounded-md aspect-square w-[100px] md:w-[176px]"
```

File: apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        />
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx:28]: question: is width attribute still needed here `width="100"` when you have `w-[100px]` class?

**Review by Szymon-dziewonski — APPROVED** [review]
looking good from my side :) !

File: apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx (hunk 2 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx:28]: I think it's better to have them as some tools like lighthouse check if they are added

File: apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const [reviewCharacterLimit] = useState(5000);
+  const ratingLabelId = useId();
+
+  const reviewIsAboveLimit = review.length > reviewCharacterLimit;
+  const reviewCharsCount = reviewCharacterLimit - review.length;
+
+  return (
+    <div className="max-w-[376px] md:max-w-[768px]">
+      <h3 className="font-bold py-2 pl-4 pr-3 typography-headline-4">Leave a review</h3>
+      <form className="grid grid-cols-[100px_1fr] py-2 px-4 gap-4 md:grid-cols-[176px_1fr] grid-rows-[100px_1fr] md:grid-rows-[28px_1fr] items-center md:items-start">
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx:21]: I'm wondering if we should add `onSubmit` handler. It should be clear to add it as we use button type submit, but it's still an extra step to do for anyone that copy this block. Do you think we could add the submit handler?

Comment (justyna-13) [reply apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx:21]: sure I can add it

Comment (justyna-13) [reply apps/preview/next/pages/showcases/RatingForms/ProductRatingWithReview.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2882/commits/bfa77ea519f54c56d6623ed3748d54a6c69b18af

**Review by dkacper — APPROVED** [review]
LGTM ⭐

**Pull request merged**